cmd/nerdctl/completion/completion.go

```diff
@@ -124,9 +124,13 @@ func NetworkNames(cmd *cobra.Command, exclude []string) ([]string, cobra.ShellCo
 	if err != nil {
 		return nil, cobra.ShellCompDirectiveError
 	}
-	for netName := range netConfigs {
+	for netName, network := range netConfigs {
 		if _, ok := excludeMap[netName]; !ok {
 			candidates = append(candidates, netName)
+			if *network.NerdctlID != "" {
```

Comment:
As far as I can tell, this is the only case where removing ids from the map would have an impact, so, maintaining the same behavior as before here.